app/src/org/commcare/android/util/AndroidInstanceInitializer.java

```diff
@@ -61,6 +61,10 @@ public String getVersionString() {
 
     @Override
     public String getDeviceId() {
-        return CommCareApplication._().getPhoneId();
+        String phoneId = CommCareApplication._().getPhoneId();
+        if (phoneId == null) {
+            return super.getDeviceId();
+        }
+        return phoneId;
```

Comment:
Robolectric returns `null` for imei, which in turn breaks some form metadata setup. This logic catches it `null` values and returns the call to super instead, which is `"----"`. Sorta messy, open to finding an alternative if someone thinks it is important.